runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -780,19 +791,24 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
 
    if (abortCompilation)
       {
+      enterPerClientAllocationRegion();
       if (clientOptions)
          {
          TR_Memory::jitPersistentFree(clientOptions);
          entry._clientOptions = NULL;
          }
-      if (optPlan)
-         TR_OptimizationPlan::freeOptimizationPlan(optPlan);
+
       if (_recompilationMethodInfo)
          {
          TR_Memory::jitPersistentFree(_recompilationMethodInfo);
          _recompilationMethodInfo = NULL;
          }
 
+      if (optPlan)
+         {
+         TR_OptimizationPlan::freeOptimizationPlan(optPlan);
```

Comment:
Does `freeOptimizationPlan()` know how to get to global allocator?

Reply:
Yeah, it needs to be moved outside per-client region.
I think this bug is not caught by tests because the only way we can reach the catch block with allocated optimization plan is if its allocation throws, which shouldn't happen under normal memory conditions. All other exceptions should throw before the plan is allocated so we never actually reach this code during testing.